### DISCUSSION TITLE
RLS for insert relation

### DISCUSSION THREAD

**Discussion by snax4a** [supabase/supabase#4654] [Questions]
Hi,
I have a problem figuring out how can I access the inserted values inside RLS insert check.

I have 4 tables

```sql
create table users (
  id uuid references auth.users not null primary key
);

create table facilities (
  id uuid not null primary key,
  name text not null
);

create table user_facility (
 user_id references users (id) not null primary key,
 facility_id references facilities (id) not null primary key
);

create table notes (
 id uuid not null primary key,
 title text not null,
 description text not null,
 facility_id references facilities (id) not null
);

```

In my RLS for notes table i want to access facility_id value and check if the user who is trying to insert the note is connected with that facility in user_facility table.

I thought I could access access inserted data with NEW variable like it is in triggers but it looks like there is no such option in RLS

If i could access NEW.facility_id then i could make
```sql
select 1 from user_facility where user_id = uid() and facility_id = NEW.facility_id
```

Do you guys have any suggestions how can i achieve that kind of security for Inserts? I don't want to allow everyone to insert notes for facility they dont belong to.

**Comment by hfnukal** [comment]
You can try:

```SQL
facility_id in (select uf.facility_id from user_facility uf where uf.user_id = auth.uid())
```

or define function to resolve this for you

```SQL
create or replace function get_facilities_for_user(user_id uuid)
returns setof bigint as $$
  select facility_id from user_facility where user_id = $1
$$ stable language sql security definer;

create policy "Insert notes for connected users"
on notes
for all using (
  facility_id in (select get_facilities_for_user(auth.uid()))
);
```
